bin/spice/pkg/taskhistory/taskhistory.go

```diff
@@ -90,15 +90,23 @@ func (tMs *TimeWithMilliSeconds) UnmarshalJSON(b []byte) error {
 	s := string(b)
 	s = s[1 : len(s)-1]
 
-	layout := "2006-01-02T15:04:05.999999"
-
-	t, err := time.Parse(layout, s)
-	if err != nil {
-		return err
-	}
-
-	*tMs = TimeWithMilliSeconds(t)
-	return nil
+	layouts := []string{
+        "2006-01-02T15:04:05.999999Z",
+        "2006-01-02T15:04:05.999999",
+    }
+
+	var t time.Time
+    var err error
+
+    for _, layout := range layouts {
+        t, err = time.Parse(layout, s)
+        if err == nil {
+            *tMs = TimeWithMilliSeconds(t)
+            return nil
+        }
+    }
+
+	return fmt.Errorf("cannot parse time: %s", s)
```

Comment:
```suggestion
	return fmt.Errorf("cannot parse time %s: %w", s, err)
```

Reply:
Good catch. Although I changed to `fmt.Errorf("cannot parse time: %w", err)`. 

Otherwise we would get:
```bash
error parsing response from spiced: cannot parse time 2025-09-16T18:40:50.843578Z: parsing time \"2025-09-16T18:40:50.843578Z\": extra text: \"Z\"
```
Which seems slightly excessive.

Now it's:
```bash
error parsing response from spiced: cannot parse time: parsing time \"2025-09-16T18:41:29.471243Z\": extra text: \"Z\"
```

(The error is an example, the current implementation handles this value properly)